contracts/xasset-orchestrator/src/test.rs

```diff
@@ -0,0 +1,82 @@
+#![cfg(test)]
+use crate::error::Error;
+use crate::{SorobanContract__, SorobanContract__Client};
+
+use loam_sdk::soroban_sdk::{testutils::Address as _, Address, Env};
+use loam_sdk::soroban_sdk::{String, Symbol};
+
+fn create_orchestrator_contract<'a>(e: &Env) -> SorobanContract__Client<'a> {
+    let orchestrator = SorobanContract__Client::new(e, &e.register(SorobanContract__, ()));
+    let admin: Address = Address::generate(e);
+    let _ = orchestrator.try_admin_set(&admin);
+    orchestrator.init(&Address::generate(e), &Address::generate(e));
+    orchestrator
+}
+
+#[test]
+fn test_orchestrator() {
+    let e = Env::default();
+    e.mock_all_auths();
+
+    // Create test address to use as arguments
+    let test_address = Address::generate(&e);
+
+    // Create an orchestrator contract
+    let orchestrator = create_orchestrator_contract(&e);
+
+    // Initialize the orchestrator with the test contract address
+    let try_deploy_result = orchestrator.try_deploy_asset_contract(
+        &test_address,
+        &Symbol::new(&e, "XLM"),
+        &100,
+        &String::from_str(&e, "XLM"),
+        &String::from_str(&e, "XUSD"),
+        &6,
+        &100,
+    );
+    assert!(try_deploy_result.is_ok());
+    let deploy_result = try_deploy_result.unwrap().unwrap();
+
+    // get_asset_contract with a non-existent asset symbol
+    let invalid_symbol = String::from_str(&e, "NOASSET");
+    let invalid_result = orchestrator.try_get_asset_contract(&invalid_symbol);
+    assert!(invalid_result.is_err());
```

Comment:
This would fail below.
```suggestion
```

Reply:
I'm unsure of the intention your comment/the change? I thought "I expect it to be an error" "this is the error I expect it to be" feel like good test steps

Reply:
My point is that unwrapping the error will panic if this is not true. So it's like testing it twice.

Reply:
Sort of, I think this is a philosophy difference. If it fails here, it would say "hey, I expected this to be an error" versus panicking when trying to unwrap and error if left to later. Would rather leave it.